azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -16,39 +16,275 @@
+         waitForOperationCompletion(addToSubnetId, operationDescription);
+      }
+
+      logger.debug("Creating a cloud service with name '%s', label '%s' in location '%s'", name, name, location);
+      String createCloudServiceRequestId = api.getCloudServiceApi().createWithLabelInLocation(name, name, location);
+      waitForOperationCompletion(createCloudServiceRequestId, format("Create cloud service (%s)", name));
+
+      final OSImage.Type os = template.getImage().getOperatingSystem().getFamily().equals(OsFamily.WINDOWS) ? OSImage.Type.WINDOWS : OSImage.Type.LINUX;
+      Set<ExternalEndpoint> externalEndpoints = Sets.newHashSet();
+      for (int inboundPort : inboundPorts) {
+         externalEndpoints.add(ExternalEndpoint.inboundTcpToLocalPort(inboundPort, inboundPort));
+      }
+      final DeploymentParams params = DeploymentParams.builder()
+              .name(name)
+              .os(os)
+              .username(loginUser)
+              .password(loginPassword)
+              .sourceImageName(template.getImage().getId())
+              .mediaLink(createMediaLink(storageServiceAccountName, name))
+              .size(RoleSize.Type.fromString(template.getHardware().getName()))
+              .externalEndpoints(externalEndpoints)
+              .subnetName(subnetName)
+              .virtualNetworkName(virtualNetworkName)
+              .build();
+      logger.debug("Creating a deployment with params '%s' ...", params);
+      String createDeploymentRequestId = api.getDeploymentApiForService(name).create(params);
+      waitForOperationCompletion(createDeploymentRequestId, format("Create deployment with params (%s)", params));
+
+      if (!retry(new Predicate<String>() {
+         public boolean apply(String name) {
+            return FluentIterable.from(api.getDeploymentApiForService(name).get(name).roleInstanceList())
+                    .allMatch(new Predicate<RoleInstance>() {
+                       @Override
+                       public boolean apply(RoleInstance input) {
+                          return input != null && input.instanceStatus() == READY_ROLE;
+                       }
+                    });
+         }
+      }, 30 * 60, 1, SECONDS).apply(name)) {
+         logger.warn("Instances %s of %s has not reached the status %s within %sms so it will be destroyed.",
+                 Iterables.toString(api.getDeploymentApiForService(name).get(name).roleInstanceList()), name,
+                 READY_ROLE, Long.toString(operationTimeout));
+         api.getDeploymentApiForService(group).delete(name);
+         api.getCloudServiceApi().delete(name);
+         throw new IllegalStateException(format("Deployment %s is being destroyed as its instanceStatus didn't reach " +
+                 "status %s after %ss. Please, try by increasing `jclouds.azure.operation-timeout` and " +
+                 " try again", name, READY_ROLE, 30 * 60));
+      }
```

Comment:
The jclouds ComputeService will already perform the active wait until the node is in state RUNNING. Is it really necessary to manually wait for the node being ready here? can't you just return the Deployment without waiting?